extensions/security/iprange/src/org/exist/security/realm/iprange/IPRangeServlet.java

```diff
@@ -63,12 +63,15 @@ protected void doPost(final HttpServletRequest request, final HttpServletRespons
 
 
         // Get reverse proxy header when available, otherwise use regular IP address
-        String ipAddress = request.getHeader("X-Forwarded-For");
-        if (ipAddress == null) {
+        String ipAddress = request.getHeader("X-Forwarded-For").replaceAll("\\s","");
```

Comment:
I am afraid you run the risk of a NPE here as `ServletRequest#getHeader(String)` can return `null` if the header is not present - http://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletRequest.html#getHeader(java.lang.String)

Perhaps instead consider something like:

```java
final String ipAddress =
    Optional.ofNullable(request.getHeader("X-Forwarded-For"))
    .map(str -> str.replaceAll("\\s", ""))
    .filter(str -> !str.isEmpty)
    .map(str -> str.split(","))
    .filter(strs -> strs.length > 1)
    .map(strs -> strs[strs.length - 1]
    .orElseGet(() -> request.getRemoteAddr())
```